### PR TITLE
Refactor ApiClient to Kotlin

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Refactor `ApiClient` and supporting files to Kotlin
 - `ApiClient` is intended for internal use only

 ### Checklist

 ~- [ ] Added a changelog entry~

- @sshropshire

### CODE REVIEW

File: BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    callback.onResult(JSONObject(responseBody), null)
+                    braintreeClient.sendAnalyticsEvent("card.graphql.tokenization.success")
+                } catch (exception: JSONException) {
+                    braintreeClient.sendAnalyticsEvent("card.graphql.tokenization.failure")
+                    callback.onResult(null, exception)
+                }
+            } else {
+                braintreeClient.sendAnalyticsEvent("card.graphql.tokenization.failure")
+                callback.onResult(null, httpError)
+            }
```
Comment (jcnoriega) [inline BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt:27]: suggestion/question: for simple `if not null { ... } else { ... }` we could replace with `jsonBody?.also { ... } ?: { ... }`. Think its worth it? reference: https://stackoverflow.com/a/45878199

File: BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            braintreeClient.sendPOST(url, body) { responseBody, httpError ->
+                if (responseBody != null) {
+                    try {
+                        callback.onResult(JSONObject(responseBody), null)
+                    } catch (exception: JSONException) {
+                        callback.onResult(null, exception)
+                    }
+                } else {
+                    callback.onResult(null, httpError)
+                }
```
Comment (jcnoriega) [inline BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt:46]: same as above

File: BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt (hunk 1 above)
Comment (jcnoriega) [reply BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt:27]: i figured as well that when we start with coroutines, some major changes are going to be implemented here. Whatever you feel like!

**Review by jcnoriega — APPROVED** [review]

File: BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt (hunk 1 above)
Comment (sshropshire) [reply BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt:27]: Would you be up for doing a PR into this branch with the change? This pattern will come up a lot once we start refactoring the clients. It'll be nice to lock in on a Kotlin strategy.

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**